packages/eas-cli/src/submit/ios/IosSubmitCommand.ts

```diff
@@ -172,12 +172,18 @@ export default class IosSubmitCommand {
       });
     }
 
-    // interpret this to mean the user had some intention of passing in ASC Api key
     if (ascApiKeyPath || ascApiKeyIssuerId || ascApiKeyId) {
       Log.warn(`ascApiKeyPath, ascApiKeyIssuerId and ascApiKeyId must all be defined in eas.json`);
-      return result({
-        sourceType: AscApiKeySourceType.prompt,
-      });
+
+      // in interactive mode, interpret this to mean the user had some intention of passing in ASC Api key
+      // skip in non-interactive mode with warning
+      if (!this.ctx.nonInteractive) {
+        return result({
+          sourceType: AscApiKeySourceType.prompt,
+        });
+      } else {
+        Log.warn("Proceeding because you've run this command in non-interactive mode.");
```

Comment:
this is better than the actual current behaviour, but i'd prefer if we throw an error instead